llvm/test/Transforms/LoopVectorize/AArch64/simple_early_exit.ll

```diff
@@ -0,0 +1,1969 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
+; RUN: opt -S < %s -p loop-vectorize -mattr=+sve -debug-only=loop-vectorize \
```

Comment:
There's nothing target specific in this test at the moment, can it be moved out so we have test coverage for the legality checks independent of the target?

Reply:
Sure, seems reasonable. We will need a few SVE specific tests once we start vectorising these loops just to ensure the cost model decides it's worth vectorising, but that's fine!